src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -356,6 +386,17 @@ def recordInCouch(self, jobs, newstate, oldstate, updatesummary=False):
                                 "archivestatus": archStatus,
                                 "fwjr": jsonFWJR,
                                 "type": "fwjr"}
+                if countDocs >= self.fwjrdatabase.getQueueSize():
+                    try:
+                        self.fwjrdatabase.commit(callback=discardConflictingDocument)
+                    except CouchRequestTooLargeError as exc:
+                        msg = "Failed to commit bulk of framework job report to CouchDB."
```

Comment:
I doubt CouchDB has rollback functionality, in other words here the behavior of commit is unpredictable. For instance, if large document will be first one to insert the entire chunk will fail, but if it is a last one I think the first documents will be successfully injected, no?

Reply:
I just chatted with CouchDB experts and they say that the behavior of `_bulk_docs` is that each document can succeed or fail independently. 
In other words, the correct way to know what succeeded or not is by parsing the response body and identifying what succeeded or not, more information in: https://docs.couchdb.org/en/stable/api/database/bulk-api.html#updating-documents-in-bulk

This goes beyond the scope of this PR though, and this PR does not change the current behavior. We need to address that in a different GH issue.

Reply:
Agreed! It seems to be a widespreaded lack of check in this module. Updating in my next commit.